### DISCUSSION TITLE
Just a thought: Make A Glitch project

### DISCUSSION THREAD

**Discussion by everythingability** [hyperdiv/hyperdiv#14] [Ideas]
Hi all.

I'm a hobby python-er... but in the past I've used Django and Flask but in my role teaching coding, they seemed to me to overkill in terms of having to learn so much OTHER stuff to python to get a webserver running.

I hacked a Glitch project together and it seems to work beautifully, though I bet you could create a better starter project than I could so people can learn the basics of hyperdiv.

https://glitch.com/edit/#!/my-hyperdiv-example

Cheers

Tom

**Comment by mariusnita** [comment]
Hey Tom, this is really cool! Thanks for sharing!

**Comment by mariusnita** [comment]
One small thought about this part of the code:
```py
button = hd.button("Click Me")
if button.clicked:
    hd.text("You clicked me")
```
Notice that the text "You clicked me" disappears when something else updates on the page. This is because clicks are transient. To make the text persist you would use state:
```py
state = hd.state(was_clicked=False)
if button.clicked:
    state.was_clicked = True
if state.was_clicked:
    hd.text("You clicked me.")
```

I'll check out Glitch, thanks for the pointer.

**Reply by everythingability** [reply]
Ooh thanks...
